cardano-cli/src/Cardano/CLI/Run/Friendly.hs

```diff
@@ -149,29 +153,52 @@ friendlyTxOut (TxOut addr amount mdatum) =
               , "amount" .= friendlyTxOutValue amount
               ]
 
-    AddressInEra (ShelleyAddressInEra sbe) saddr@(ShelleyAddress net cred stake) ->
-      let preAlonzo :: [Aeson.Pair]
-          preAlonzo =
-            [ "address era" .= Aeson.String "Shelley"
-            , "network" .= net
-            , "payment credential" .= cred
-            , "stake reference" .= friendlyStakeReference stake
-            , "address" .= serialiseAddress saddr
-            , "amount" .= friendlyTxOutValue amount
-            ]
-          datum :: ShelleyBasedEra era -> [Aeson.Pair]
-          datum ShelleyBasedEraShelley = []
-          datum ShelleyBasedEraAllegra = []
-          datum ShelleyBasedEraMary = []
-          datum ShelleyBasedEraAlonzo = ["datum" .= renderDatum mdatum]
-      in object $ preAlonzo ++ datum sbe
+    AddressInEra
+      (ShelleyAddressInEra sbe)
+      saddr@(ShelleyAddress net cred stake) ->
+        object $ preAlonzo ++ datum
+     where
+      preAlonzo =
+        [ "address era" .= Aeson.String "Shelley"
+        , "network" .= net
+        , "payment credential" .= cred
+        , "stake reference" .= friendlyStakeReference stake
+        , "address" .= serialiseAddress saddr
+        , "amount" .= friendlyTxOutValue amount
+        ]
+      datum =
+        [ friendlyDatum mdatum
+        | isJust $ scriptDataSupportedInEra $ shelleyBasedToCardanoEra sbe
+        ]
+
+friendlyDatum :: TxOutDatum CtxTx era -> Aeson.Pair
+friendlyDatum = \case
+  TxOutDatumNone -> "datum" .= Null
+  TxOutDatumHash _ h -> "datum hash" .= String (serialiseToRawBytesHexText h)
+  TxOutDatum _ sData -> "datum" .= conv sData
   where
-   renderDatum :: TxOutDatum CtxTx era -> Aeson.Value
-   renderDatum TxOutDatumNone = Aeson.Null
-   renderDatum (TxOutDatumHash _ h) =
-     Aeson.String $ serialiseToRawBytesHexText h
-   renderDatum (TxOutDatum _ sData) =
-     scriptDataToJson ScriptDataJsonDetailedSchema sData
+
+    -- | 'ScriptData' does not support text, only byte strings.
+    -- 'ScriptData.scriptDataToJson' formats byte strings as Base16 blobs.
+    -- So here we try to display text data where byte strings look like text.
+    conv :: ScriptData -> Aeson.Value
```

Comment:
There is a lot of repetition of code here from `scriptDataToJson`. I suggest using `scriptDataToJson`, pattern matching on the `Value`s of interest (`Object` in your case) and dealing with the Map keys as you see fit.

Reply:
That will be an extra 0x-base-16 encoding followed by extra 0x-base-16 decoding. Is it better?

Reply:
I don't understand what you mean. Can you clarify?

Reply:
I don't understand what you mean. Can you clarify?

Reply:
So this is what I have in mind:
```
conv :: ScriptData -> ScriptDataJsonSchema -> Aeson.Value
conv sd schema = 
  case scriptDataToJson schema sd of 
    Aeson.Array a -> ...
    ...
```